lib/datadog/statsd.rb

```diff
@@ -337,35 +343,69 @@ def format_event(title, text, opts={})
     private_constant :NEW_LINE, :ESC_NEW_LINE, :COMMA, :BLANK, :PIPE, :DOT,
       :DOUBLE_COLON, :UNDERSCORE
 
-    def escape_event_content(msg)
-      msg.gsub NEW_LINE, ESC_NEW_LINE
+    def escape_event_content!(msg)
+      msg.gsub!(NEW_LINE, ESC_NEW_LINE) || msg
     end
 
-    def escape_tag_content(tag)
-      remove_pipes(tag).gsub COMMA, BLANK
+    def escape_tag_content!(tag)
```

Comment:
> We often freeze tags. I believe this will cause an exception to raise for frozen strings. Should we use gsub for frozen strings or something else?

Ah, yes. I'll add a test for that and handle frozen strings differently. 

> Or to put it other way: it's bad form to modify the arguments that the user is passing to the library: even if they're not frozen, the user won't expect those arguments to change. The ! methods can only be used on strings we've allocated inside the library.

Yeah, I definitely felt weird about that and noticed that @aroben `dup`-ed
 the stat in [here](https://github.com/github/statsd-ruby/pull/16/files#diff-179756856cd1fe6003145224a9f4f979R139), but it also seemed counterproductive to `dup`  the arguments when the point is to reduce allocations. I'll add the `dup`s and run the benchmark again to see what the difference is 👍